packages/@react-aria/gridlist/src/useGridListItem.ts

```diff
@@ -178,7 +178,7 @@ export function useGridListItem<T>(props: AriaGridListItemOptions, state: ListSt
     }
   };
 
-  let linkProps = getSyntheticLinkProps(node.props);
+  let linkProps = itemStates.hasAction ? getSyntheticLinkProps(node.props) : {};
```

Comment:
Means `data-href` only applies when nothing is selected (in which case the action applies). This allows the cursor to change to pointer when it's actually a link.